packages/core/src/llm-core/chat/app.ts

```diff
@@ -222,15 +222,31 @@ export class ChatInterface {
         }
 
         // Process response
-        this.ctx.parallel(
-            'chatluna/after-chat',
-            arg.conversationId,
-            arg.message,
-            displayResponse as AIMessage,
-            { ...arg.variables, chatCount: this._chatCount },
-            this,
-            arg.session
-        )
+        try {
+            await this.ctx.parallel(
+                'chatluna/after-chat',
+                arg.conversationId,
+                arg.message,
+                displayResponse as AIMessage,
+                { ...arg.variables, chatCount: this._chatCount },
+                this,
+                arg.session
+            )
+        } catch (error) {
+            logger.warn(
+                'Unhandled error in after-chat hook, ignored to keep process alive'
+            )
+            logger.warn(error)
+            await this.ctx.parallel(
```

Comment:
Use the `handleChatError` method, and add a `throwError: boolean` parameter to it.